### ISSUE TITLE
Exception within TextField.onSubmitted gets swallowed

### ISSUE THREAD

**Issue by MichaelRFairhurst** [flutter/flutter#13076]
## Steps to Reproduce

Create a TextField with an `onSubmitted` callback which throws:

```
              onSubmitted: (String str) {                                        
                print('in onSubmitted');                                         
                throw 'diediediediedie'; 
```

And from the app, submit the text.

The exception is swallowed:

```
Reloaded 1 of 457 libraries in 417ms.
I/flutter ( 3892): in onSubmitted
```

That's the end.

Had no issues with flutter analyze, either.

## Flutter Doctor

```
[✓] Flutter (on Linux, locale en_US.UTF-8, channel alpha)
    • Flutter at /home/mfairhurst/dart/flutter
    • Framework revision e8aa40eddd (4 weeks ago), 2017-10-17 15:42:40 -0700
    • Engine revision 7c4142808c
    • Tools Dart version 1.25.0-dev.11.0

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).
      If Android SDK has been installed to a custom location, set $ANDROID_HOME to that location.

[✗] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).

[-] IntelliJ IDEA Ultimate Edition (version 2016.2)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    • Dart plugin version 162.2485
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins
    ✗ This install is older than the minimum recommended version of 2017.1.0.

[-] IntelliJ IDEA Ultimate Edition (version 2017.2)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    • Dart plugin version 172.2791
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins

[-] IntelliJ IDEA Ultimate Edition (version 2016.3)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    • Dart plugin version 163.8651
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins
    ✗ This install is older than the minimum recommended version of 2017.1.0.

[✓] Connected devices
    • None
```

**Comment by zoechi** [comment]
Still an issue 
```dart
import 'package:flutter/material.dart';

void main() {
  FlutterError.onError = (e) {
    print('FlutterError handler: $e');
  };
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text(
              'foo text',
            ),
            TextField(
              onSubmitted: (String str) {
                print('in onSubmitted');
                throw 'diediediediedie';
              },
            )
          ],
        ),
      ),
    );
  }
}
```

```
Performing hot restart...
Syncing files to device Android SDK built for x86...
Restarted application in 2,825ms.
I/flutter ( 9036): in onSubmitted
```

```
[✓] Flutter (Channel dev, v1.1.9, on Mac OS X 10.14.1 18B75, locale en-AT)
```

**Comment by TahaTesser** [comment]
<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel dev, v1.18.0-6.0.pre, on Mac OS X 10.15.4 19E287, locale
    en-GB)
    • Flutter version 1.18.0-6.0.pre at /Users/taha/Code/flutter_dev
    • Framework revision 84c84fb249 (3 days ago), 2020-04-20 21:35:01 -0400
    • Engine revision 2bde4f0ae4
    • Dart version 2.9.0 (build 2.9.0-1.0.dev a12c36dd97)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/taha/Code/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/taha/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.
[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.9.1
[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
[✓] VS Code (version 1.44.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.9.1
[✓] Connected device (4 available)
    • Android SDK built for x86 • emulator-5554 • android-x86    • Android 10
      (API 29) (emulator)
    • macOS                     • macOS         • darwin-x64     • Mac OS X
      10.15.4 19E287
    • Chrome                    • chrome        • web-javascript • Google Chrome
      81.0.4044.122
    • Web Server                • web-server    • web-javascript • Flutter Tools
• No issues found!
```

</details>

**Comment by YeungKC** [comment]
This issue cannot be reproduced on the master branch, it may have been fixed, please retest.

<details>
  <summary>log</summary>
  flutter: in onSubmitted

════════ Exception caught by widgets ═══════════════════════════════════════════════════════════════
The following message was thrown while calling onSubmitted for TextInputAction.done:
diediediediedie

When the exception was thrown, this was the stack: 
#0      _MyHomePageState.build.<anonymous closure> (package:foo/main.dart:46:17)
#1      EditableTextState._finalizeEditing (package:flutter/src/widgets/editable_text.dart:1898:27)
#2      EditableTextState.performAction (package:flutter/src/widgets/editable_text.dart:1758:9)
#3      TextInput._handleTextInputInvocation (package:flutter/src/services/text_input.dart:1278:37)
#4      MethodChannel._handleAsMethodCall (package:flutter/src/services/platform_channel.dart:436:55)
...
════════════════════════════════════════════════════════════════════════════════════════════════════
</details>

cc @TahaTesser

**Comment by YeungKC** [comment]
This issue does not reproduce on the master branch and has been fixed at #66851, please close the issue.

<details>
  <summary>log</summary>
  flutter: in onSubmitted

════════ Exception caught by widgets ═══════════════════════════════════════════════════════════════
The following message was thrown while calling onSubmitted for TextInputAction.done:
diediediediedie

When the exception was thrown, this was the stack: 
#0      _MyHomePageState.build.<anonymous closure> (package:foo/main.dart:46:17)
#1      EditableTextState._finalizeEditing (package:flutter/src/widgets/editable_text.dart:1898:27)
#2      EditableTextState.performAction (package:flutter/src/widgets/editable_text.dart:1758:9)
#3      TextInput._handleTextInputInvocation (package:flutter/src/services/text_input.dart:1278:37)
#4      MethodChannel._handleAsMethodCall (package:flutter/src/services/platform_channel.dart:436:55)
...
════════════════════════════════════════════════════════════════════════════════════════════════════
</details>

cc @TahaTesser @justinmc
